kintone-purify.js

```diff
@@ -0,0 +1,19 @@
+(function() {
+	const events = ['app.record.create.submit', 
+					'app.record.edit.submit', 
+					'app.record.index.edit.submit'];
+	const fields_to_clean = ['product'];
+	kintone.events.on(events, function(event) {
+		const record = event.record;
+		for (let field in record) {
+			if (record.hasOwnProperty(field)) {
+				if (fields_to_clean.includes(field)) {
```

Comment:
It's up to you if you fix the code, but you can make this simpler with `Object.keys()` and `Array.prototype.forEach()`.

```js
Object.keys(record).forEach(field => {
  if (fieldsToClean.includes(field)) {
    // ...
  }
})
```